stytch/b2b/sessions.go

```diff
@@ -323,7 +331,25 @@ func (c *SessionsClient) AuthenticateJWTLocal(
 		return nil, sessions.ErrJWTTooOld
 	}
 
-	return marshalJWTIntoSession(claims)
+	memberSession, err := marshalJWTIntoSession(claims)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal JWT into session: %w", err)
+	}
+
+	var policy *rbac.Policy
+	if authorizationCheck != nil {
+		policy, err = c.PolicyCache.Get(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get cached policy: %w", err)
+		}
+	}
+
+	err = shared.PerformAuthorizationCheck(policy, claims.Roles, memberSession.OrganizationID, authorizationCheck)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
[boulder] This check belongs inside the `if authorizationCheck != nil` block -- otherwise you're about to do an auth check against a `nil` policy

Reply:
Ahh, I read below that `shared.PerformAuthorizationCheck` will check if the authCheck is nil, but I still think it'd be safer to put this _inside_ the block above to be crystal clear.